### PR TITLE
Docs: Added the documentation

### LINKED ISSUE #109: Docs: Add the docs regarding the architecture of the app
### Description

As the project continues to evolve, it's crucial to have comprehensive documentation that provides insights into the app's architecture. This documentation will not only serve as a valuable resource for the current contributors but also for newcomers who want to understand how the app is structured and organized.

I'm planning to add these key points to the documentation:-

- Overview of the App's High-Level Architecture
- Directory Structure
- Key Modules and Components
- Data Flow
- Technology Stack
- Code Patterns and Best Practices
- Development Guidelines

Additionally, I'll be including visual diagrams, and links to relevant resources to enhance the comprehensibility of the documentation. 

_I would like to work on this issue, so please assign it to me._

### PR DESCRIPTION
**Pull request by superiorsd10**
### Description
I've added comprehensive documentation covering the important aspects of the application. I've added the following points to the documentation:- 

- GetX Pattern
- Database Schema
- Installation & Setup
- User Interface & Features
- Contribution Guidelines
- Future Plans
- Community
- Flutter

### Proposed Changes
- Changed the `README.md` file

## Screenshots

_No response_

## Checklist

- [ ] Tests have been added or updated to cover the changes
- [x] Documentation has been updated to reflect the changes
- [ ] Code follows the established coding style guidelines
- [ ] All tests are passing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,55 +7,297 @@ This project aims to build a non-conventional alarm clock with smart features su
...
+- [GetX Pattern](#getx-pattern)
+- [Database Schema](#database-schema)
+- [Installation & Setup](#installation-setup)
+- [User Interface & Features](#user-interface-features)
+- [Contribution Guidelines](#contribution-guidelines)
+- [Future Plans](#future-plans)
+- [Community](#community)
+- [Flutter](#flutter)
+
+## GetX Pattern
```
Comment (MarkisDev) [inline README.md:20]: We should also mention about using get_cli to create new pages to make everything easier

Also, we need to briefly touch upon what each of those files do in general (controller, view etc).

File: README.md [hunk 2]
```diff
@@ -6,55 +7,297 @@ This project aims to build a non-conventional alarm clock with smart features su
...
+The "Ultimate Alarm Clock" project employs the GetX pattern for state management. The GetX pattern is a popular state management solution in the Flutter ecosystem, known for its simplicity, efficiency, and developer-friendly approach. It simplifies the process of managing the state of a Flutter application and helps in building reactive and performant user interfaces.
+
+To learn more about the GetX, you can read the documentation [_here_](https://chornthorn.github.io/getx-docs/).
+
+## Database Schema
+
+The "Ultimate Alarm Clock" project utilizes multiple databases for different purposes. These databases include Firebase Firestore, ISAR, and Flutter Secure Storage. Each database serves a distinct role in managing various aspects of the application. Below we provide an overview of the schema for each of these databases. 
+
+### Firebase Firestore
+
```
Comment (MarkisDev) [inline README.md:31]: Should probably make it clearer why we have 2 databases. Firestore is necessary for shared alarms, IsarDb is necessary for offline alarms

Comment (MarkisDev) [reply README.md:31]: Also, we need to talk about how the shared alarms is setup using Firebase since I think that's the most confusing part

**Review by MarkisDev — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (superiorsd10) [reply README.md:20]: Okay, I'll be adding this.

File: README.md (hunk 2 above)
Comment (superiorsd10) [reply README.md:31]: Right, I should have touched upon that part. I'll be doing that now.

**Comment by superiorsd10** [comment]
Hey @MarkisDev 👋
I've made the necessary additions. Please review them and tell me if any changes or more addition are required. 

Thank you :)

**Pull request merged**